client/components/promo-section/promo-card/cta.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+
+import React, { FunctionComponent } from 'react';
+import { connect } from 'react-redux';
+import { partialRight } from 'lodash';
+
+/**
+ * Internal depencies
+ */
+import Button from 'components/button';
+import ActionPanelCta from 'components/action-panel/cta';
+import { hasFeature } from 'state/sites/plans/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { URL } from 'types';
+
+type ClickCallback = () => void;
+
+export interface CtaButton {
+	text: string;
+	action: URL | ClickCallback;
+}
+
+export type Cta =
+	| CtaButton
+	| {
+			feature: string;
+			upgradeButton: CtaButton;
+			defaultButton: CtaButton;
+			activatedButton?: CtaButton;
+	  };
+
+interface ConnectedProps {
+	hasPlanFeature: boolean;
+}
+
+export interface Props {
+	cta: Cta;
+	learnMoreLink?: string;
+	isPrimary?: boolean;
+}
+
+function isCtaButton( cta: Cta ): cta is CtaButton {
+	return undefined !== ( cta as CtaButton ).text;
+}
+
+function buttonProps( button: CtaButton, isPrimary: boolean ) {
+	return {
+		className: 'promo-card__cta-button',
+		primary: isPrimary,
+		[ typeof button.action === 'string' ? 'href' : 'onClick' ]: button.action,
+	};
+}
+const PromoCardCta: FunctionComponent< Props & ConnectedProps > = ( {
+	cta,
+	learnMoreLink,
+	isPrimary,
+	hasPlanFeature,
+} ) => {
+	const ctaBtnProps = partialRight( buttonProps, true === isPrimary );
+	let ctaBtn;
+
+	if ( isCtaButton( cta ) ) {
+		ctaBtn = <Button { ...ctaBtnProps( cta ) }>{ cta.text }</Button>;
+	} else {
+		ctaBtn = hasPlanFeature ? (
+			<Button { ...ctaBtnProps( cta.defaultButton ) }>{ cta.defaultButton.text }</Button>
+		) : (
+			<Button { ...ctaBtnProps( cta.upgradeButton ) }>{ cta.upgradeButton.text }</Button>
+		);
+	}
+	return (
+		<ActionPanelCta>
+			{ ctaBtn }
+			{ learnMoreLink && (
+				<Button borderless className="promo-card__cta-learn-more" href="{ learnMoreLink }">
+					Learn More
```

Comment:
These lines have the same issue I fixed in
https://github.com/Automattic/wp-calypso/commit/6857e9388916a7511f3f778e425424cea448e620

The `learnMoreLink` between braces and the untranslatable `Learn more`. Perhaps we should merge this first and then the other? Or merge the other and fix the eventual conflict.